src/main/scala/org/constellation/p2p/DataResolver.scala

```diff
@@ -15,15 +15,15 @@ class DataResolver extends StrictLogging {
 
   def resolveMessagesDefaults(
     hash: String,
-    priorityClient: Option[PeerApiClient] = None
-  )(implicit apiTimeout: Duration = 3.seconds, dao: DAO): IO[Option[ChannelMessageMetadata]] =
+    priorityClient: IO[Option[PeerApiClient]] = IO.pure(None)
+  )(implicit apiTimeout: Duration = 3.seconds, dao: DAO): IO[ChannelMessageMetadata] =
     resolveMessages(hash, getReadyPeers(dao), priorityClient)
 
   def resolveMessages(
     hash: String,
-    pool: Iterable[PeerApiClient],
-    priorityClient: Option[PeerApiClient]
-  )(implicit apiTimeout: Duration = 3.seconds, dao: DAO): IO[Option[ChannelMessageMetadata]] = {
+    pool: IO[List[PeerApiClient]],
```

Comment:
you should avoid such things and passing `IO` here instead of a value.
you can use `flatMap` in place where you actually need to use it instead of passing IO here